test/test_torch.py

```diff
@@ -12151,6 +12151,25 @@ def test_ceil_out_mismatch(self, device):
         b = torch.randn(1, device=device)
         self.assertRaises(RuntimeError, lambda: torch.ceil(a, out=b))
 
+    # Test for unary ops appropriate for floating-type output from integral and boolean inputs
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_intbool_to_float_upcasting(self, device, dtype):
+        # Not all the functions are appropriate for this upcasting, so only test for the list below
+        op_list = ["acos", "asin", "ceil", "expm1", "floor", "log", "log10", "log1p", "log10", "log1p",
+                   "log2", "sin", "sinh", "sqrt", "trunc", "atan", "cos", "cosh", "exp", "tan",
+                   "tanh"]
+        # Make sure that the ops promote to floating default types
+        # Test both float32 and double (float64) dtypes
+        saved_dtype = torch.get_default_dtype()
```

Comment:
It might be worth defining a context manager to save and restore the dtype, so the dtype will be restored even if something in the loop below throws an error.

Reply:
Resolved by https://github.com/pytorch/pytorch/pull/33322/#discussion_r388504876